gnocchi/amqp1d.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (c) 2015 eNovance
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import print_function
+import itertools
+import ujson
+import uuid
+
+import daiquiri
+from oslo_config import cfg
+
+import six
+
+from gnocchi import incoming
+from gnocchi import indexer
+from gnocchi import service
+
+from gnocchi import utils
+
+from proton.handlers import MessagingHandler
+from proton.reactor import Container
+
+LOG = daiquiri.getLogger(__name__)
+
+
+class CollectdStats(object):
```

Comment:
Ok I am calling it as CollectDStats because you can use amqp1 to  consume messages from different source. So the idea was if we enhance this to include stats  from statsd via AMQP1.0 then you will have StatsDStats class etc. The format is different here. So let me know if should I rename to just stats for now?

Reply:
I think I get it now. Will remove collectd and just call stats. I need to add some ability to add format of the  message  like json message schema something like that. Will do some R&D on that.

Reply:
I think you can use `voluptuous` library like we do for the rest of the REST API to validate the data structure. I think it's ok for now to only have one schema support, so don't overthink it :)